packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -767,7 +774,19 @@ export class AnchorSet implements Listenable<AnchorSetRootEvents>, AnchorLocator
 						continue;
 					}
 					emittedEvents?.push(event);
-					node.events.emit(event, node);
+					if (event === "childrenChangedAfterBatch") {
+						const fieldKeys = this.bufferedEvents
+							.filter((e) => e.node === node && e.event === event)
+							.map((e) => e.changedField as FieldKey);
+						// For array nodes, the field key will always be an empty string.
+						// In that case we want to emit the event with an empty set, instead of a set with the empty string as only member.
+						const changedFields = fieldKeys.every((x) => x === "")
```

Comment:
Isn't it possible for an object or map node to have a field whose key is ""? I'm concerned this code will filter that out.

Reply:
I'm wondering if instead of doing this filtering here, it would be better to do it downstream (like in TreeNodeKernel) where you acutally have access to the schema.

Reply:
If you do end up keeping the `""`, replace it with the constant `EmptyKey` for consistency.

Reply:
Would that be for code clarity/maintainability? If we do that then I'd motion we only do that in the array case, and not the case map/object with a "" key. Otherwise we'd run the risk of making things really confusing for users if we ever change the value for `EmptyKey`.

Reply:
Nothing more to it than the "DRY" advice to factor out literals/constants. I agree we should only do it for arrays, because as-is it seems broken, and the correct solution here probably doesn't use `""` or `EmptyKey` at all.

Reply:
I'll see about doing this in treeNodeKernel.

Reply:
Done in latest, will be pushed soon. Of note, we don't have a `isArrayNodeSchema()` function like we do a `isObjectNodeSchema()`. I wanted to go that route but adding it didn't seem trivial (we have an [ObjectNodeSchema type](https://github.com/microsoft/FluidFramework/blob/966906a03490daa5a914030b37342abb8267c12d/packages/dds/tree/src/simple-tree/objectNodeTypes.ts#L25) that the function type-asserts, but no equivalent intermediate type for array nodes (see [here](https://github.com/microsoft/FluidFramework/blob/966906a03490daa5a914030b37342abb8267c12d/packages/dds/tree/src/simple-tree/arrayNode.ts#L886-L895)), so I went with a simple condition on `nodeSchema.kind === NodeKind.Array` which was enough since I didn't need to use any array-specific features for what we're doing here.